azure-mgmt-appservice/src/main/java/com/microsoft/azure/management/appservice/implementation/AppServiceManager.java

```diff
@@ -202,4 +214,155 @@ public FunctionApps functionApps() {
         }
         return functionApps;
     }
+
+    /**
+     *
+     * @return Returns the latest Windows runtime stacks
+     */
+    public AppServiceRuntimes latestWindowsRuntimes() {
+        Hashtable<String, RuntimeVersion> variousRuntimes = new Hashtable<>();
+        variousRuntimes.put(JavaVersion.COMPONENT_NAME.toLowerCase(), JavaVersion.OFF);
+        variousRuntimes.put(NodeVersion.COMPONENT_NAME.toLowerCase(), NodeVersion.OFF);
+        variousRuntimes.put(WebContainer.COMPONENT_NAME.toLowerCase(), WebContainer.OFF);
+        variousRuntimes.put(PythonVersion.COMPONENT_NAME.toLowerCase(), PythonVersion.OFF);
+        variousRuntimes.put(PhpVersion.COMPONENT_NAME.toLowerCase(), PhpVersion.OFF);
+        variousRuntimes.put(NetFrameworkVersion.COMPONENT_NAME.toLowerCase(), NetFrameworkVersion.OFF);
+
+        Iterator<ApplicationStackInner> stackIter = this.innerManagementClient.providers().getAvailableStacks("Windows").iterator();
+        while (stackIter.hasNext()) {
+            ApplicationStackInner stack = stackIter.next();
+
+            RuntimeVersion runtime = variousRuntimes.get(stack.name().toLowerCase());
+            if (runtime != null) {
+                runtime.parseApplicationStackInner(stack);
+            } else {
+                //throw new FileNotFoundException("Runtime " + stack.name() + " not supported");
+            }
+        }
+
+        AppServiceRuntimesImpl runtimes = new AppServiceRuntimesImpl();
+
+        runtimes.withJavaVersions(JavaVersion.values());
+        runtimes.withNetFrameworkVersions(NetFrameworkVersion.values());
+        runtimes.withPhpVersions(PhpVersion.values());
+        runtimes.withPythonVersions(PythonVersion.values());
+        runtimes.withWebContainers(WebContainer.values());
+        runtimes.withNodeVersions(NodeVersion.values());
+
+        return runtimes;
+    }
+
+    /**
+     * The runtimes data to be consumed by the user.
+     */
+    public class AppServiceRuntimesImpl implements AppServiceRuntimes {
```

Comment:
keep it package private so that user won't be able to access this impl.

Reply:
will do.